client/src/api/Rank/rank_crud.ts

```diff
@@ -0,0 +1,35 @@
+import {
+	TopPostsRes,
+	TopCommentsRes,
+	TopActivitiesRes,
+} from "../../../../nestjs-migration/dist/rank/dto/rank-results.dto";
+
+export const fetchPostRank = async () => {
+	const response = await fetch("http://localhost:3000/api/rank/posts");
+	if (!response.ok) {
+		throw new Error("Network response was not ok");
+	}
+
+	const data: TopPostsRes[] = await response.json();
+	return data;
+};
+
+export const fetchCommentRank = async () => {
+	const response = await fetch("http://localhost:3000/api/rank/comments");
+	if (!response.ok) {
+		throw new Error("Network response was not ok");
+	}
+
+	const data: TopCommentsRes[] = await response.json();
+	return data;
+};
+
+export const fetchActivitiesRank = async () => {
+	const response = await fetch("http://localhost:3000/api/rank/activities");
+	if (!response.ok) {
+		throw new Error("Network response was not ok");
+	}
+
+	const data: TopActivitiesRes[] = await response.json();
+	return data;
+};
```

Comment:
`nit` 
fetchPostRank, fetchCommentRank, fetchActivitiesRank 3개의 함수 구조가 거의 동일하기 때문에 공통 함수로 추출해서 중복 코드를 줄일 수 있을 것 같습니다. 

```suggestion
const fetchRankData = async <T>(endpoint: string): Promise<T> => {
  const response = await fetch(`${API_URL}${endpoint}`);
  if (!response.ok) {
    throw new Error(`Failed to fetch ${endpoint}`);
  }

  const data = await response.json();
  return data;
};

// 사용 예시
export const fetchPostRank = () => fetchRankData<TopPostsRes[]>("/api/rank/posts");
export const fetchCommentRank = () => fetchRankData<TopCommentsRes[]>("/api/rank/comments");
export const fetchActivitiesRank = () => fetchRankData<TopActivitiesRes[]>("/api/rank/activities");
```